app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -248,8 +248,16 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
         // code paths consistent between the two.
         if (event.keyCode == KeyEvent.KEYCODE_BACK ||
                 event.keyCode == KeyEvent.KEYCODE_DEL) return super.dispatchKeyEvent(event)
+
         serviceLocator.cursorEventRepo.pushKeyEvent(event)
 
+        serviceLocator.cursorController.maybeToMotionEvent(event)?.let {
```

Comment:
This is called redundantly inside `handleKeyEvent`. I feel like we should return two values or a sealed class from `handleKeyEvent` so that we keep a single entry point. While having two entry points can increase clarity in some ways, I dislike that it adds complexity to the call contract where two methods need to be called instead of one.

Reply:
That was the original implementation before you asked me to change it to this one.  I'm fine with either, do you have a strong preference one way or the other?

Reply:
I don't remember asking for that or why I would – I would prefer a single entry point.

Reply:
`handleKeyEvent` returned a sealed class that looked something like `HandleResponse(val wasConsumed: Boolean, motionEvent: MotionEvent?)`.  It wasn't great code.  I prefer to have a single entry point as well, but both implementations have drawbacks.